nemo/core/classes/modelPT.py

```diff
@@ -1839,11 +1916,17 @@ def on_train_start(self):
     def on_train_batch_start(self, batch: Any, batch_idx: int, unused: int = 0) -> Optional[int]:
         """PyTorch Lightning hook:
         https://pytorch-lightning.readthedocs.io/en/stable/common/lightning_module.html#on-train-batch-start
-        We use it here to enable nsys profiling and dynamic freezing.
+        We use it here to enable profiling and dynamic freezing.
         """
-
-        # nsys profiling
         if self.device.type == 'cuda':
+            if hasattr(self, '_chakra_profile_enabled'):
+                if self._chakra_profile_enabled and not self._chakra_profile_in_progress:
+                    if self.trainer.global_step >= self._chakra_profile_start_step and self.trainer.global_step < self._chakra_profile_end_step:
```

Comment:
based on line 1796 (`if self._chakra_profile_end_step >= self._chakra_profile_start_step`), we want to support profiling for just a single step, right? so we should do `self.trainer.global_step <= self._chakra_profile_end_step`

Reply:
Hi @maanug-nv , Thanks for the reviewing, I missed the global step will increase in batch_end when I adding this line, I have tested the use case of capturing single iteration, and confirmed you are right, fixed it

Reply:
fixed it in the latest commit, resolve this conversation